modules/standard/Random.chpl

```diff
@@ -2869,6 +2865,19 @@ module Random {
         }
       }
 
+      /* Randomly shuffle an ND array. */
+      proc ref shuffle(ref arr: [?D] )
+        where D.isRectangular() && D.rank > 1
+      {
+        // Fisher-Yates shuffle
+        for i in 0..#D.sizeAs(D.idxType) by -1 {
+          const k = D.orderToIndex(PCGRandomStreamPrivate_getNext_noLock(this.eltType, 0:this.eltType, i:this.eltType));
```

Comment:
Can you split this line up? Maybe factor out `PCGRandomStreamPrivate_getNext_noLock(this.eltType, 0:this.eltType, i:this.eltType)`?